### PR TITLE
Welcome update

### PR DESCRIPTION
**Pull request by benfoley**
Whoosh. 

This combines :

- DEV_MODE env var feature to show a special toolbar for developers
- new flow from welcome page
- show training logs in GUI
- misc style things
- more to come probably..stay tuned

### CODE REVIEW

**Comment by nicklambourne** [comment]
😂 
Screen Shot 2021-03-04 at 1 29 34 pm

**Comment by nicklambourne** [comment]
This may take me a while this evening.

**Comment by benfoley** [comment]
yeah, there's a fair bit here. I was trying to keep it to smaller ones but the merging of branches was doing my head in..

**Review by nicklambourne — APPROVED** [review]
Some general questions and style suggestions, but lgtm

File: Dockerfile [hunk 1]
```diff
@@ -173,7 +173,7 @@ WORKDIR /
 
 # Elpis
 RUN pwd
-RUN git clone --depth=1 https://github.com/CoEDL/elpis.git
+RUN git clone --single-branch --branch welcome_update --depth=1 https://github.com/CoEDL/elpis.git
```
[the remark this answers is not in the document]
Comment (benfoley) [reply Dockerfile:176]: Correct - i put that in to test the dockerfile. must have accidentally pushed it

File: elpis/endpoints/model.py [hunk 2]
```diff
@@ -78,11 +77,14 @@ def list_existing():
...
                 'name': model['name'],
                 'dataset_name': model['dataset_name'],
+                'engine_name': model['engine_name'],
                 'pron_dict_name': model['pron_dict_name'],
                 'status': model['status'],
                 'results': fake_results
                 } for model in interface.list_models_verbose()]
     }
+    # interface.list_models_verbose()
+    # data = {"list": []}
```
[the remark this answers is not in the document]
Comment (benfoley) [reply elpis/endpoints/model.py:87]: Nope. Leftover scraps. Removed.

File: elpis/endpoints/model.py [hunk 3]
```diff
@@ -135,6 +137,7 @@ def status():
         "status": model.status,
         "stage_status": model.stage_status
     }
+    print(data)
```
[the remark this answers is not in the document]
Comment (benfoley) [reply elpis/endpoints/model.py:140]: Do you mean spacing in the code or in the print output?

File: elpis/engines/common/objects/interface.py [hunk 4]
```diff
@@ -225,34 +223,37 @@ def get_model(self, mname):
...
+                with self.models_path.joinpath(hash_dir, "model.json").open() as fin:
                     name = json.load(fin)['name']
                     models.append(name)
         return models
 
     def list_models_verbose(self):
-        if self.engine is None:
-            raise RuntimeError("Engine must be set to list models")
+        # if self.engine is None:
+        #     raise RuntimeError("Engine must be set to list models")
```
[the remark this answers is not in the document]
Comment (benfoley) [reply elpis/engines/common/objects/interface.py:238]: Previous design was that models couldn't be listed unless engine was specified. Removing this gives us scope to list models independent of engines, and then we can load the engine based on the selected model.

File: elpis/endpoints/model.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicklambourne) [reply elpis/endpoints/model.py:138]: In the code

File: elpis/engines/common/objects/interface.py [hunk 5]
```diff
@@ -225,34 +223,37 @@ def get_model(self, mname):
         return m
 
     def list_models(self):
-        if self.engine is None:
-            raise RuntimeError("Engine must be set to list models")
+        # if self.engine is None:
+        #     raise RuntimeError("Engine must be set to list models")
```
[the remark this answers is not in the document]
Comment (benfoley) [reply elpis/engines/common/objects/interface.py]: nope, see next comment

**Pull request merged**